### PR TITLE
Add timeout to flush buffer if last flush was longer than x seconds a…

### PR DESCRIPTION
**Pull request by c0by**
…go. Set with --buffer_timeout, disabled by default, can be used in conjunction with --buffer_size. Tested with confluent reader.

### CODE REVIEW

**Comment by mre** [comment]
Thanks @c0by. Looks good. Can we add a unit test for the worker timeout?
Should go into [test_woker.py](https://github.com/mre/kafka-influxdb/blob/master/kafka_influxdb/tests/test_worker.py). Feel free to ask for support. 😃

**Comment by c0by** [comment]
I have added the unit test. Could use some input though, as KeyboardInterrupt is called at the end the buffer will be written even if the timeout fails...

File: kafka_influxdb/tests/test_worker.py [hunk 1]
```diff
@@ -32,6 +34,24 @@ def read(self):
...
+    """
+    A reader that writes half the messages then times out
+    """
+    def __init__(self, message, num_messages, timeout):
+        self.message = message
+        self.num_messages = num_messages
+        self.timeout = timeout
+
+    def read(self):
+        # Yield the first half of messages
```
Comment (mre) [inline kafka_influxdb/tests/test_worker.py]: Can you remove this comment? It's a bit misleading ;-)

Comment (c0by) [inline kafka_influxdb/tests/test_worker.py]: Done :)

**Comment by mre** [comment]
Thanks for the unit test. 😃 
As for your last comment, I don't I think I understand the problem yet. Are you saying that it's hard to test if the timeout really works because of the keyboard interrupt that comes right after?
If so, why not remove the `raise KeyboardInterrupt`, since all you want to test is timeout? The keyboard interrupt is tested separately.

**Comment by c0by** [comment]
Yes, that's basically it. Removing `raise KeyboardInterrupt` leaves the test in a infinite loop, from `Worker.consume() `. So I need to raise something else to break out after the test, any thoughts what's best to use?

**Comment by mre** [comment]
I see. Does it work if we add a timeout decorator like in the other examples?

```python
@timeout(0.2)
def test_buffer_timeout(self):
  # same as before
```

**Comment by c0by** [comment]
Hmm, no doesn't look like it, when the timeout decorator terminates the process `self.assertTrue(self.writer.write.called)` never gets called, so no test is done.

Could we add an additional `except` to the loop in `Worker.consume()` that doesn't call `Worker.flush()` when raised?

**Comment by c0by** [comment]
Are you happy this is an acceptable fix?

**Comment by mre** [comment]
Sure thing. Thanks for your patience @c0by. Highly appreciate it. 😊
Will merge and update the changelog.

**Pull request merged**